Homeworks/Homework_3/script.py

```diff
@@ -0,0 +1,369 @@
+
+    :param destination_folder_path: str, the path of the folder to set as the new working directory.
+    :return: tuple, containing the old working directory path and the new working directory path.
+    """
+    old_work_directory = os.getcwd()
+    os.chdir(destination_folder_path)
+    new_work_directory = destination_folder_path
+    return old_work_directory, new_work_directory
+
+
+def move_csv_file(old_dir, new_dir, csv_file_name):
+    """
+    Move a CSV file from the old directory to the new directory.
+
+    :param old_dir: str, the path of the current directory containing the CSV file.
+    :param new_dir: str, the path of the destination directory where the CSV file will be moved.
+    :param csv_file_name: str, the name of the CSV file to be moved.
+    :return: None
+    """
+
+    os.rename(os.path.join(old_dir, csv_file_name), os.path.join(new_dir, csv_file_name))
+
+
+def rearrange_data(old_csv_data):
+    """
+    Rearrange data into format 'decades : {countries : [users_data]}'
+
+    :param old_csv_data: list, data from the csv file in the old format
+    :return: dict, data which was rearranged from list format to the dict format
+    """
+    new_data = {}
+    for row in old_csv_data:
+        decade = f'{row["dob.date"][-2]}0-th'
+        country = row['location.country']
+        new_data.setdefault(decade, {})
+        new_data[decade].setdefault(country, [])
+        new_data[decade][country].append(row)
+    return new_data
+
+
+def create_file_name(rearranged_data, folder_decade, folder_country):
+    """
+    Create a specific name for the CSV file based on user's maximum age, average number of years of registration,
+    and the most common id.name.
+
+    :param rearranged_data: list, containing all data about users, organized by decade and country.
+    :param folder_decade: str, the name of the folder representing the decade.
+    :param folder_country: str, the name of the folder representing the country.
+    :return: str, the name of the CSV file based on the provided data.
+    """
+    users_age = [int(user['dob.age']) for user in rearranged_data[folder_decade][folder_country]]
+    users_registered_age = [int(user['registered.age']) for user in rearranged_data[folder_decade][folder_country]]
+    users_id_name = [user['id.name'] for user in rearranged_data[folder_decade][folder_country]]
+
+    return f'{max(users_age)}_' \
+           f'{sum(users_registered_age) / len(users_registered_age)}_' \
+           f'{Counter(users_id_name).most_common(1)[0][0]}'
+
+
+def create_sub_folders(rearranged_data):
+    """
+    Create folders with csv files based on the new data format
+
+    :param rearranged_data: dict, data on the structure of which
+                            folders are created and csv files into folders are filled
+    """
+    for decade in rearranged_data:
+        decade_path = os.path.join(os.getcwd(), decade)
+        os.makedirs(decade_path)
```

Comment:
you don't need this two lines.  the next makedirs can create a full hierarchy for you